package/MDAnalysis/coordinates/TRJ.py

```diff
@@ -544,7 +544,11 @@ def __init__(self, filename, n_atoms=None, mmap=None, **kwargs):
 
         # checks for not-implemented features (other units would need to be
         # hacked into MDAnalysis.units)
-        self._verify_units(self.trjfile.variables['time'].units, 'picosecond')
+        try:
+            self._verify_units(self.trjfile.variables['time'].units, 'picosecond')
+            self.has_time = True
+        except KeyError:
+            self.has_time = False
```

Comment:
I realise I said it was nearly good to go, but I just realised that we probably want to warn users when this happens (so they are at least aware that `times` isn't going to be something they should expect.

If you're keen to get this merged, I'm happy to merge this as-is @DrDomenicoMarson and then have another PR add the warning.

Reply:
If you mean that in line 551 we should warn the users something like
"No time information found in the trajectory, 'times' will be set as an increasing index"?

If that's the case I can add the warning now and just have it all in this PR!

Reply:
Yes ideally that'd be it + an additional `with pytest.warns` test alongside it 🙏🏽

Reply:
Done, I hope in a proper way!